src/agent/scm/git.ts

```diff
@@ -103,19 +113,37 @@ export class GitScmProvider extends scmm.ScmProvider {
         return Q(0)
         .then((code: number) => {
 	        if (!this.enlistmentExists()) {
-	        	return this.gitw.clone(giturl, true, folder, gopt);
+	        	return this.gitw.clone(giturl, true, folder, gopt).then((result: number) => {
```

Comment:
at first I thought you had multiple returns but then I noticed the .then at the end of the line.   Can you left align that so it's clearer what's going on?